controllers/storagecluster/provider_server.go

```diff
@@ -202,6 +202,12 @@ func GetProviderAPIServerDeployment(instance *ocsv1.StorageCluster) *appsv1.Depl
 							Name:    "ocs-provider-api-server",
 							Image:   os.Getenv(providerAPIServerImage),
 							Command: []string{"/usr/local/bin/provider-api"},
+							Env: []corev1.EnvVar{
+								{
+									Name:  "WATCH_NAMESPACE",
+									Value: os.Getenv("WATCH_NAMESPACE"),
```

Comment:
There is a `GetWatchNamespace` method to get this namespace. It has error validations as well. Suggesting to use it here to get the `WATCH_NAMESPACE` value.

Reply:
I am doing the same as a func for the other env, added this also to the list.

Reply:
so this approach ends up fetching the env variable data twice.  Once in `LookupEnv` and again in `Getenv`. Suggesting to have a single call to fetch the env variables.

Reply:
Actually, I am doing that for a reason `createDeployment` func returns an error where I am checking it for the first time and returning an error if ENV does not exist. `GetProviderAPIServerDeployment` func does not return the error it just returns deployment only that's why not checking in that and I do not want to send ENVS as the argument.

Reply:
ok.